internal/clients/http/azure/azure_client.go

```diff
@@ -10,61 +10,55 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions"
 	"github.com/RHEnVision/provisioning-backend/internal/clients"
 	"github.com/RHEnVision/provisioning-backend/internal/config"
-	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
-	"github.com/rs/zerolog"
 )
 
-type azureClient struct {
+type client struct {
 	subscriptionID string
 	credential     *azidentity.ClientSecretCredential
 }
 
 func init() {
-	clients.GetAzureClient = newAzureClient
+	clients.GetCustomerAzureClient = newCustomerClient
```

Comment:
Because of the two different (clients) types. For Azure it is literally a different type, for EC2 it can be the same client (with just different configuration), is it is technically possible to have a single interface for EC2, but for consistency, I would recommend to do the same approach for all the three.